File: php/features.php
<?php
declare(strict_types=1);

// Motivational SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class MotivationalFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new MotivationalBaseFeature();
            case "test":
                return new MotivationalTestFeature();
            default:
                return new MotivationalBaseFeature();
        }
    }
}

File: py/features.py
# Motivational SDK feature factory

from feature.base_feature import MotivationalBaseFeature
from feature.test_feature import MotivationalTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MotivationalBaseFeature(),
        "test": lambda: MotivationalTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
